File: Semana_2/Cypress/JavaScript/npm/curso/e2e/cypress/pages/separated/open-account.actions.js
const { OPEN_ACCOUNT_LOCATORS } = require('./open-account.locators');
const { capture } = require('../../support/evidence');

class OpenAccountActions {
  openSavingsAccount() {
    cy.get(OPEN_ACCOUNT_LOCATORS.accountType).select('SAVINGS');
    capture('parabank-separated-select-savings');
    cy.get(OPEN_ACCOUNT_LOCATORS.fromAccount)
      .find('option')
      .its('length')
      .should('be.greaterThan', 0);

    cy.get(OPEN_ACCOUNT_LOCATORS.openAccountButton).click();
    capture('parabank-separated-click-open-account');

    return cy
      .get(OPEN_ACCOUNT_LOCATORS.newAccountId, { timeout: 10000 })
      .should(($id) => {
        expect($id.text().trim()).to.match(/^\d+$/);
      })
      .invoke('text')
      .then((text) => {
        const accountId = text.trim();
        capture('parabank-separated-account-created');
        return cy.wrap(accountId);
      });
  }
}

module.exports = { OpenAccountActions };

File: Semana_2/Selenium/Java/maven/curso/e2e/src/test/java/com/curso/selenium/pages/separated/OpenAccountActions.java
package com.curso.selenium.pages.separated;

import com.curso.selenium.evidence.EvidenceSupport;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class OpenAccountActions {
    private final WebDriver driver;
    private final WebDriverWait wait;

    public OpenAccountActions(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    public String openSavingsAccount() {
        wait.until(ExpectedConditions.numberOfElementsToBeMoreThan(OpenAccountLocators.FROM_ACCOUNT_OPTIONS, 0));
        new Select(driver.findElement(OpenAccountLocators.ACCOUNT_TYPE_SELECT)).selectByVisibleText("SAVINGS");
        EvidenceSupport.capture(driver, "parabank-separated-select-savings");
        driver.findElement(OpenAccountLocators.OPEN_ACCOUNT_BUTTON).click();
        EvidenceSupport.capture(driver, "parabank-separated-click-open-account");
        String accountId = wait.until(ExpectedConditions.visibilityOfElementLocated(OpenAccountLocators.NEW_ACCOUNT_ID)).getText();
        EvidenceSupport.capture(driver, "parabank-separated-account-created");
        return accountId;
    }
}
